galerautils/src/gu_asio_stream_engine.cpp

```diff
@@ -75,8 +76,9 @@ class AsioTcpStreamEngine : public gu::AsioStreamEngine
         {
             return op_result{success, static_cast<size_t>(bytes_written) };
         }
-        else if (errno == EAGAIN || errno == EWOULDBLOCK)
+        else if (errno == EAGAIN)
```

Comment:
Why this change?

Reply:
Because they are the same, and generate warnings with the warning settings in the server code. I've added static_asserts to ensure that they are indeed the same on all platforms where we build pxc.

Reply:
I mean if it is guaranteed that they are the same, I'm fine with this, but I see other places in the code (not Galera), where there is a distinction between the two.
For example viosocket.cc solves this problem is this way:
```
#if SOCKET_EAGAIN == SOCKET_EWOULDBLOCK
    if (error != SOCKET_EAGAIN)
#else
    if (error != SOCKET_EAGAIN && error != SOCKET_EWOULDBLOCK)
#endif
```

Reply:
Yes, that's why I also added a static assert. I did a search about this change, and found this:
https://stackoverflow.com/questions/7003234/which-systems-define-eagain-and-ewouldblock-as-different-values/7003379#7003379

Which explains that they are only different on some old unix systems (SysV, HPUX, old AIX version, OS/390).

I doubt we'll ever build pxc in any of those.